acceptance: TestDockerCLI_test_demo_node_cmds failed [decomission timeout; unexpected (?) liveness errors after a node is decommissioned]

acceptance.TestDockerCLI_test_demo_node_cmds [failed](https://mesolite.cluster.engflow.com/invocations/default/6899d2b5-dfa7-4acb-a84a-88db69e00851?testReportRun=1&testReportShard=13&testReportAttempt=1#targets-Ly9wa2cvYWNjZXB0YW5jZTphY2NlcHRhbmNlX3Rlc3Q=) on master @ [826ee929e41e2d1a1634a18348a8495c5ec5218d](https://github.com/cockroachdb/cockroach/commits/826ee929e41e2d1a1634a18348a8495c5ec5218d):


```
=== RUN   TestDockerCLI_test_demo_node_cmds
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestDockerCLI_test_demo_node_cmds767116314
    test_log_scope.go:76: use -show-logs to present logs inline
    generated_cli_test.go:133: -- test log scope end --
test logs left over in: outputs.zip/logTestDockerCLI_test_demo_node_cmds767116314
--- FAIL: TestDockerCLI_test_demo_node_cmds (69.19s)
=== RUN   TestDockerCLI_test_demo_node_cmds/test_demo_node_cmds.tcl
    util_docker.go:256: non-zero exit code: 1
--- FAIL: TestDockerCLI_test_demo_node_cmds/test_demo_node_cmds.tcl (60.52s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=1</code>
 - <code>shard=13</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #144952 acceptance: TestDockerCLI_test_demo_node_cmds failed [C-test-failure O-robot P-2 T-sql-foundations X-infra-flake branch-release-25.2 no-test-failure-activity]
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestDockerCLI_test_demo_node_cmds.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-51283

acceptance.TestDockerCLI_test_demo_node_cmds [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_Acceptance/19964049?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_Acceptance/19964049?buildTab=artifacts#/) on master @ [4fa39749d45d53e496fdc53fab3a1eae0290e63e](https://github.com/cockroachdb/cockroach/commits/4fa39749d45d53e496fdc53fab3a1eae0290e63e):


```
=== RUN   TestDockerCLI_test_demo_node_cmds
    test_log_scope.go:165: test logs captured to: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/acceptance/_tmp/0f973bfae5165b427f0ee0932d9c6ec9/logTestDockerCLI_test_demo_node_cmds1848381080
    test_log_scope.go:76: use -show-logs to present logs inline
    generated_cli_test.go:133: -- test log scope end --
test logs left over in: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/acceptance/_tmp/0f973bfae5165b427f0ee0932d9c6ec9/logTestDockerCLI_test_demo_node_cmds1848381080
--- FAIL: TestDockerCLI_test_demo_node_cmds (70.95s)
=== RUN   TestDockerCLI_test_demo_node_cmds/test_demo_node_cmds.tcl
    util_docker.go:256: non-zero exit code: 1
--- FAIL: TestDockerCLI_test_demo_node_cmds/test_demo_node_cmds.tcl (60.86s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestDockerCLI_test_demo_node_cmds.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

acceptance.TestDockerCLI_test_demo_node_cmds [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_Acceptance/19988145?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_Acceptance/19988145?buildTab=artifacts#/) on master @ [561b230087d39b5c635eaa0c0234bade962e7c15](https://github.com/cockroachdb/cockroach/commits/561b230087d39b5c635eaa0c0234bade962e7c15):


```
=== RUN   TestDockerCLI_test_demo_node_cmds
    test_log_scope.go:165: test logs captured to: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/acceptance/_tmp/0f973bfae5165b427f0ee0932d9c6ec9/logTestDockerCLI_test_demo_node_cmds4215298388
    test_log_scope.go:76: use -show-logs to present logs inline
    generated_cli_test.go:133: -- test log scope end --
test logs left over in: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/acceptance/_tmp/0f973bfae5165b427f0ee0932d9c6ec9/logTestDockerCLI_test_demo_node_cmds4215298388
--- FAIL: TestDockerCLI_test_demo_node_cmds (69.83s)
=== RUN   TestDockerCLI_test_demo_node_cmds/test_demo_node_cmds.tcl
    util_docker.go:256: non-zero exit code: 1
--- FAIL: TestDockerCLI_test_demo_node_cmds/test_demo_node_cmds.tcl (59.62s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestDockerCLI_test_demo_node_cmds.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

acceptance.TestDockerCLI_test_demo_node_cmds [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_Acceptance/20207419?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_Acceptance/20207419?buildTab=artifacts#/) on master @ [ac400e78b0ff3f3e5c09960a7927e11862f74f43](https://github.com/cockroachdb/cockroach/commits/ac400e78b0ff3f3e5c09960a7927e11862f74f43):


```
=== RUN   TestDockerCLI_test_demo_node_cmds
    test_log_scope.go:171: test logs captured to: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/acceptance/_tmp/0f973bfae5165b427f0ee0932d9c6ec9/logTestDockerCLI_test_demo_node_cmds1639570028
    test_log_scope.go:82: use -show-logs to present logs inline
    generated_cli_test.go:133: -- test log scope end --
--- FAIL: TestDockerCLI_test_demo_node_cmds (70.54s)
=== RUN   TestDockerCLI_test_demo_node_cmds/test_demo_node_cmds.tcl
    util_docker.go:256: non-zero exit code: 1
--- FAIL: TestDockerCLI_test_demo_node_cmds/test_demo_node_cmds.tcl (58.72s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #150423 acceptance: TestDockerCLI_test_demo_node_cmds failed [C-test-failure O-robot T-sql-foundations branch-release-25.2 release-blocker]
- #149151 acceptance: TestDockerCLI_test_demo_node_cmds failed [decomission timeout] [A-testing C-bug C-test-failure O-robot P-3 T-kv branch-release-25.3]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestDockerCLI_test_demo_node_cmds.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

acceptance.TestDockerCLI_test_demo_node_cmds [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_Acceptance/20244024?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_Acceptance/20244024?buildTab=artifacts#/) on master @ [e0bc803646d69704a66ae69129ef7e7a3b7059f6](https://github.com/cockroachdb/cockroach/commits/e0bc803646d69704a66ae69129ef7e7a3b7059f6):


```
=== RUN   TestDockerCLI_test_demo_node_cmds
    test_log_scope.go:171: test logs captured to: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/acceptance/_tmp/0f973bfae5165b427f0ee0932d9c6ec9/logTestDockerCLI_test_demo_node_cmds478537826
    test_log_scope.go:82: use -show-logs to present logs inline
    generated_cli_test.go:133: -- test log scope end --
--- FAIL: TestDockerCLI_test_demo_node_cmds (66.73s)
=== RUN   TestDockerCLI_test_demo_node_cmds/test_demo_node_cmds.tcl
    util_docker.go:256: non-zero exit code: 1
--- FAIL: TestDockerCLI_test_demo_node_cmds/test_demo_node_cmds.tcl (57.97s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #150423 acceptance: TestDockerCLI_test_demo_node_cmds failed [C-test-failure O-robot T-sql-foundations branch-release-25.2 release-blocker]
- #149151 acceptance: TestDockerCLI_test_demo_node_cmds failed [decomission timeout] [A-testing C-bug C-test-failure O-robot P-3 T-kv branch-release-25.3]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestDockerCLI_test_demo_node_cmds.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

acceptance.TestDockerCLI_test_demo_node_cmds [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_Acceptance/20298541?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_Acceptance/20298541?buildTab=artifacts#/) on master @ [0c343c645fa5f9e986c28e389afd312d4dd89184](https://github.com/cockroachdb/cockroach/commits/0c343c645fa5f9e986c28e389afd312d4dd89184):


```
=== RUN   TestDockerCLI_test_demo_node_cmds
    test_log_scope.go:171: test logs captured to: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/acceptance/_tmp/0f973bfae5165b427f0ee0932d9c6ec9/logTestDockerCLI_test_demo_node_cmds3975020944
    test_log_scope.go:82: use -show-logs to present logs inline
    generated_cli_test.go:133: -- test log scope end --
test logs left over in: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/acceptance/_tmp/0f973bfae5165b427f0ee0932d9c6ec9/logTestDockerCLI_test_demo_node_cmds3975020944
--- FAIL: TestDockerCLI_test_demo_node_cmds (70.27s)
=== RUN   TestDockerCLI_test_demo_node_cmds/test_demo_node_cmds.tcl
    util_docker.go:256: non-zero exit code: 1
--- FAIL: TestDockerCLI_test_demo_node_cmds/test_demo_node_cmds.tcl (58.68s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #150423 acceptance: TestDockerCLI_test_demo_node_cmds failed [C-test-failure O-robot P-2 T-sql-foundations branch-release-25.2]
- #149151 acceptance: TestDockerCLI_test_demo_node_cmds failed [decomission timeout] [A-testing C-bug C-test-failure O-robot P-3 T-kv branch-release-25.3]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestDockerCLI_test_demo_node_cmds.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

--- a/pkg/cli/interactive_tests/test_demo_node_cmds.tcl
+++ b/pkg/cli/interactive_tests/test_demo_node_cmds.tcl
@@ -65,7 +65,7 @@ send "\\demo restart 3\r"
 eexpect "node 3 has been restarted"
 eexpect "defaultdb>"
 
-set timeout 1
+set timeout 2
 set stmt "select node_id, draining, membership from crdb_internal.kv_node_liveness ORDER BY node_id;\r"
 send $stmt
 expect {
@@ -89,6 +89,26 @@ eexpect "3 |    f     | active"
 eexpect "4 |    f     | active"
 eexpect "5 |    f     | active"
 eexpect "defaultdb>"
+
+# Wait for the liveness range to have the default 5 voters. If its replication
+# factor is too low, shutting down the node below can cause it to lose quorum
+# and stall the decommissioning command (example: #147867).
+set timeout 2
+set stmt "select range_id, array_length(voting_replicas,1) from crdb_internal.ranges where range_id=2;\r"
+send $stmt
+expect {
+    "2 |            5" {
+        puts "\rliveness range has 5 voters"
+    }
+    timeout {
+        puts "\rliveness range does not yet have 5 voters"
+        sleep 2
+        send $stmt
+        exp_continue
+    }
+}
+# Reset timeout back to 45 to match common.tcl.
+set timeout 45
 
 # Try decommissioning commands
 send "\\demo decommission 4\r"

